client.go

```diff
@@ -256,7 +256,9 @@ func (r *marathonClient) apiCall(method, uri string, body, result interface{}) e
 
 		// step: Create the endpoint url
 		url := fmt.Sprintf("%s/%s", member, uri)
-		if r.config.DCOSToken != "" {
+		if r.config.DCOSToken != "" && r.config.DCOSPath != "" {
```

Comment:
Instead of special-casing here, how about using `NewClient(config)` to set `r.config.DCOSPath` to the default `/marathon` (or `marathon`) if `r.config.DCOSToken` is non-empty? I'm assuming here that `DCOSToken` must always be set in DCOS Marathon; please let me know if that's not a requirement.

This would also enable us to use DCOSPath elsewhere and always have a valid value.